### PR TITLE
pylint check for dunder del AKA "__del__"

### PR DESCRIPTION
**Pull request by cmcmarrow**
checker that will raise warning if dunder del is used

https://github.com/saltstack/salt/issues/52031

GPG: Yes

### CODE REVIEW

**Review by twangboy — APPROVED** [review]

**Review by waynew — APPROVED** [review]
💥 💥 💥 💥 💥 💥 💥 💥 
Yes, please!

File: saltpylint/dunder_del.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    priority = -1
+    msgs = {
+        WARNING_CODE: (
+            '"__del__" is not allowed!',
+            'no-dunder-del',
+            '"__del__" is not allowed! A "with" block could be a good solution'
+        ),
+    }
+
+    def __init__(self, linter=None):
```
Comment (s0undt3ch) [inline saltpylint/dunder_del.py]: Do we actually need to overwrite `__init__`?

File: saltpylint/dunder_del.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        '''
+        info: setups pylint dunder del checker
+        :param linter:
+        '''
+        super(BaseChecker, self).__init__(linter)
+
+    def visit_functiondef(self, node):
```
Comment (s0undt3ch) [inline saltpylint/dunder_del.py:46]: Do we know if `visit_functiondef` is only called for classes?
We should only be checking classes.

**Review by s0undt3ch — COMMENTED** [review]
Aside from the above 2 questions, looks great!

File: saltpylint/dunder_del.py (hunk 2 above)
Comment (cmcmarrow) [reply saltpylint/dunder_del.py:46]: It checks both functions and methods. But I cant find anything like visit_methoddef.

Comment (cmcmarrow) [reply saltpylint/dunder_del.py:46]: Also no one should have a function named __del__. So I don't see why we cant check for that too.

**Pull request merged**